coriolis/tests/minion_manager/rpc/test_tasks.py

```diff
@@ -0,0 +1,2038 @@
+        mock_execute.assert_not_called()
+
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_get_minion_machine'
+    )
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_add_minion_pool_event'
+    )
+    def test_execute_with_allocation_status_and_fail_on_error(
+            self, mock_add_minion_pool_event, mock_get_minion_machine):
+        self.minion_machine.allocation_status = (
+            tasks.constants.MINION_MACHINE_STATUS_ERROR)
+        self.task._fail_on_error = True
+        mock_get_minion_machine.return_value = self.minion_machine
+
+        self.assertRaises(
+            exception.InvalidMinionMachineState, self.task.execute,
+            mock.sentinel.context, mock.sentinel.origin,
+            mock.sentinel.destination, mock.sentinel.task_info)
+
+        mock_get_minion_machine.assert_called_once_with(
+            mock.sentinel.context, self.minion_machine_id,
+            raise_if_not_found=False)
+
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_get_minion_machine'
+    )
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_add_minion_pool_event'
+    )
+    @mock.patch.object(tasks.BaseMinionManangerTask, 'execute')
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin,
+        '_set_minion_machine_allocation_status'
+    )
+    def test_execute_with_exception(
+            self, mock_set_minion_machine_allocation, mock_execute,
+            mock_add_minion_pool_event, mock_get_minion_machine):
+        mock_get_minion_machine.return_value = self.minion_machine
+        mock_execute.side_effect = CoriolisTestException
+
+        result = self.task.execute(
+            mock.sentinel.context, mock.sentinel.origin,
+            mock.sentinel.destination, mock.sentinel.task_info)
+        self.assertEqual(result['healthy'], False)
+
+        mock_get_minion_machine.assert_called_once_with(
+            mock.sentinel.context, self.minion_machine_id,
+            raise_if_not_found=False)
+        mock_set_minion_machine_allocation.assert_called_once_with(
+            mock.sentinel.context, self.minion_pool_id, self.minion_machine_id,
+            tasks.constants.MINION_MACHINE_STATUS_ERROR)
+
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_get_minion_machine'
+    )
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_add_minion_pool_event'
+    )
+    @mock.patch.object(tasks.BaseMinionManangerTask, 'execute')
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin,
+        '_set_minion_machine_allocation_status'
+    )
+    def test_execute_with_exception_and_fail_on_error(
+            self, mock_set_minion_machine_allocation, mock_execute,
+            mock_add_minion_pool_event, mock_get_minion_machine):
+        mock_get_minion_machine.return_value = self.minion_machine
+        mock_execute.side_effect = CoriolisTestException
```

Comment:
I don't get what you're testing here. You mock the tested method, make it raise and then check that the raise has occurred? 
For this case, you need to:
1. set `mock_get_minion_machine.return_value = None`;
2. remove the `mock_execute` entirely;
3. keep `_fail_on_error` on `True`;
4. Check that the call raises `InvalidMinionMachineState`.

Reply:
Nevermind this, this case was already covered. This case is also good. Thank you!